apa102/softspi.go

```diff
@@ -1,24 +1,30 @@
 package apa102
 
-import "machine"
+import (
+	"tinygo.org/x/drivers"
+	"tinygo.org/x/drivers/internal/legacy"
+)
 
 // bbSPI is a dumb bit-bang implementation of SPI protocol that is hardcoded
 // to mode 0 and ignores trying to receive data. Just enough for the APA102.
 // Note: making this unexported for now because it is probable not suitable
 // most purposes other than the APA102 package. It might be desirable to make
 // this more generic and include it in the TinyGo "machine" package instead.
 type bbSPI struct {
-	SCK   machine.Pin
-	SDO   machine.Pin
-	Delay uint32
+	SCK    drivers.PinOutput
```

Comment:
```suggestion
	SetSCK    drivers.PinOutput
```
picked this as a first example, but applies for all occurrences in this PR

reasoning:
* in go it is best-practice that get-functions not start with "get"
* the functions "Low()", "High()" are not named quite correct ("SetLow()", "SetHigh()" would be better), but are still understandable by its context e.g. "pin.Low()"
* with this in mind, for "SCK(low)" I read it as "get-state-of-SCK-for-false" and was confused, why the return value is not used

Reply:
We should also have in mind, that the "SCK" is not an attribute anymore, but a function now - maybe the name "drivers.PinOutput" does not make that clear enough?

Reply:
I'm still not convinced of this change. In Go we typically avoid setters and getters. I understand where the confusion stems from and am open to the change though yet skeptical

Reply:
> In Go we typically avoid setters and getters

this is not quite correct: https://go.dev/doc/effective_go#Getters